src/partition/ParquetInterface.cc

```diff
@@ -0,0 +1,353 @@
+
+    std::unique_ptr<parquet::arrow::FileReader> reader;
+    PARQUET_THROW_NOT_OK(parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
+
+    std::shared_ptr<parquet::FileMetaData> metadata = reader->parquet_reader()->metadata();
+    // std::cout<<"Metadata version : "<<metadata->version()<<std::endl;
+    // std::cout<<"Metadata created : "<<metadata->created_by() <<std::endl;
+    // std::cout<<"Metadata #row groups : "<<metadata->num_row_groups() <<std::endl;
+    // std::cout<<"Metadata #rows : "<<metadata->num_rows() <<std::endl;
+
+    return metadata->num_rows();
+}
+
+// Read an arrow batch, format the table acording to the config file and save it in csv format
+arrow::Status ParquetFile::ReadNextBatch_Table2CSV(void* buf, int& buffSize,
+                                                   std::vector<std::string> params) {
+    std::shared_ptr<arrow::Table> table_loc;
+
+    m_parameterNames = params;
+    arrow::Status batchStatus = ReadNextBatchTable_Formatted(table_loc);
+
+    if (!batchStatus.ok()) return arrow::Status::ExecutionError("Error while reading and formating batch");
+
+    arrow::Status status = Table2CSVBuffer(table_loc, buffSize, buf);
+
+    if (status.ok()) return arrow::Status::OK();
+
+    return arrow::Status::ExecutionError("Error while writing table to CSV buffer");
+}
+
+// Create char buffer containing the table in csv format
+arrow::Status ParquetFile::Table2CSVBuffer(std::shared_ptr<arrow::Table>& table, int& buffSize, void* buf) {
+    ARROW_ASSIGN_OR_RAISE(auto outstream, arrow::io::BufferOutputStream::Create(1 << 10));
+
+    // Options : null string, no header, no quotes around strings
+    arrow::csv::WriteOptions writeOpt = arrow::csv::WriteOptions::Defaults();
+    writeOpt.null_string = "\\N";
+    writeOpt.include_header = false;
+    writeOpt.quoting_style = arrow::csv::QuotingStyle::None;
+
+    ARROW_RETURN_NOT_OK(arrow::csv::WriteCSV(*table, writeOpt, outstream.get()));
+    ARROW_ASSIGN_OR_RAISE(auto buffer, outstream->Finish());
+
+    // auto buffer_ptr = buffer.get()->data();
+    buffSize = buffer->size();
+    std::cout << "ParquetFile::Table2CSVBuffer - buffer length : " << buffSize << " // " << m_maxBufferSize
+              << std::endl;
+
+    memcpy(buf, (void*)buffer.get()->data(), buffer->size());
+
+    //  auto myfile = std::fstream("buffer_"+std::to_string(m_batchNumber)+".csv", std::ios::out);
+    //  myfile.write((char*)&buffer_ptr[0], buffSize);
+    //  myfile.close();
+
+    return arrow::Status::OK();
+}
+
+// Read a batch from the file and format the table iaccording to the partition configuration file
+//    -> column reordering, true/false -> 1/0, remove null values, ...
+arrow::Status ParquetFile::ReadNextBatchTable_Formatted(std::shared_ptr<arrow::Table>& outputTable) {
+    arrow::Result<std::shared_ptr<arrow::RecordBatch>> maybe_batch = m_rb_reader_gbl->Next();
```

Comment:
LHS of the one-time assignments is better to be declared as `const`. It results in a safer code:
```
arrow::Result<std::shared_ptr<arrow::RecordBatch>> maybe_batch = m_rb_reader_gbl->Next();
```
Alternatively, it could be just:
```
auto const maybe_batch = m_rb_reader_gbl->Next();
```